### PR TITLE
NEW: add method as_dict. Close #11

### LINKED ISSUE #11: Add as_dict method to make it compatible with settings-vial API
The `class Settings` does not have the method `as_dict` which makes it incompatible with [SimpleSettings](https://github.com/kpn-digital/py-pkgsettings/blob/master/pkgsettings/pkgsettings.py#L9) object. 

If `pkgsettings` is used and the `ENV` variables are loaded using `settings-vial`, calling `settings.as_dict()` will raise `AttributeError`, breaking at line [31](https://github.com/kpn-digital/py-pkgsettings/blob/master/pkgsettings/pkgsettings.py#L31)

```python
settings = Settings() # pkgsettings.Settings instance
env_settings = settings_vial.Settings(env_prefix="MY_PREFIX_")
env_settings.load_env()
settings.configure(env_settings)

settings.as_dict()

File "/Users/marcos/Projects/test-with-streaming/venv/lib/python3.9/site-packages/settings_vial/settings_vial.py", line 77, in __getattr__
    raise AttributeError("{} has no attribute {}".format(self, attr))
AttributeError: <Settings {'env_prefix': 'MY_PREFIX_', 'override_prefix': None, 'override_keys_function': None, 'override_keys_reverse_lookup': False, '_config': {'ENVFILES_SECRETS': '/Users/marcos/Projects/test-with-streaming/params.env', 'ENVIRONMENT': 'localdev'}, '_override_config': {}}> has no attribute as_dict
```

### PR DESCRIPTION
**Pull request by marcosschroh**

### CODE REVIEW

File: tests/test_dotenv.py [hunk 1]
```diff
@@ -61,3 +61,9 @@ def test_prefix_gets_stripped_once_only(self, mock_envfile):
         settings.load_dotenv_file(mock_envfile)
 
         assert settings.SOME_KEY_PREFIX_REPEATED == "nested"
+
+    def test_as_dict(self, mock_envfile):
+        settings = Settings(env_prefix="PREFIX_")
+        settings.load_dotenv_file(mock_envfile)
+
+        assert settings.as_dict() == settings._config
```
Comment (mjholtkamp) [inline tests/test_dotenv.py]: Maybe nitpick, but I believe you are not testing that this is a dict (it assumes that `settings._config` is a dict I guess). What do you think about comparing it a literal dict?

File: settings_vial/settings_vial.py [hunk 2]
```diff
@@ -149,3 +149,6 @@ def _load_overrides(self):
 
         for var in vars_to_delete:
             del self._config[var]
+
+    def as_dict(self):
```
Comment (beerdeaap) [inline settings_vial/settings_vial.py:154]: I think you need to return a copy of the _conf dict here. We don't want 'outsiders' to mutate this dict right ?

File: tests/test_dotenv.py (hunk 1 above)
Comment (marcosschroh) [reply tests/test_dotenv.py]: Yes, make sense. I will add that

File: settings_vial/settings_vial.py (hunk 2 above)
Comment (marcosschroh) [reply settings_vial/settings_vial.py:154]: I think makes sense. Thanks

**Review by mjholtkamp — APPROVED** [review]

**Pull request merged**